{"version":3,"file":"NFTErrorBoundary.js","sources":["../../../src/nft/components/NFTErrorBoundary.tsx"],"sourcesContent":["import {\n  Component,\n  type ComponentType,\n  type ErrorInfo,\n  type ReactNode,\n} from 'react';\n\ntype Props = {\n  fallback?: ComponentType<{ error: Error }>;\n  children: ReactNode;\n};\n\ntype State = {\n  error: Error | null;\n};\n\nclass NFTErrorBoundary extends Component<Props, State> {\n  state: State = {\n    error: null,\n  };\n\n  static getDerivedStateFromError(error: Error): State {\n    return { error };\n  }\n\n  componentDidCatch(error: Error, errorInfo: ErrorInfo) {\n    console.error('Uncaught error:', error, errorInfo);\n  }\n\n  render() {\n    if (this.state.error) {\n      if (this.props.fallback) {\n        const Fallback = this.props.fallback;\n        return <Fallback error={this.state.error} />;\n      }\n      return <h1>Sorry, we had an unhandled error</h1>;\n    }\n\n    return this.props.children;\n  }\n}\n\nexport default NFTErrorBoundary;\n"],"names":[],"mappings":";;AAgBA,MAAM,yBAAyB,UAAwB;AAAA,EAAvD,cAAA;AAAA,UAAA,GAAA,SAAA;AACiB,SAAA,QAAA;AAAA,MACb,OAAO;AAAA,IACT;AAAA,EAAA;AAAA,EAEA,OAAO,yBAAyB,OAAqB;AACnD,WAAO,EAAE,MAAM;AAAA,EAAA;AAAA,EAGjB,kBAAkB,OAAc,WAAsB;AAC5C,YAAA,MAAM,mBAAmB,OAAO,SAAS;AAAA,EAAA;AAAA,EAGnD,SAAS;AACH,QAAA,KAAK,MAAM,OAAO;AAChB,UAAA,KAAK,MAAM,UAAU;AACjB,cAAA,WAAW,KAAK,MAAM;AAC5B,eAAQ,oBAAA,UAAA,EAAS,OAAO,KAAK,MAAM,OAAO;AAAA,MAAA;AAErC,aAAA,oBAAC,QAAG,UAAgC,mCAAA,CAAA;AAAA,IAAA;AAG7C,WAAO,KAAK,MAAM;AAAA,EAAA;AAEtB;"}